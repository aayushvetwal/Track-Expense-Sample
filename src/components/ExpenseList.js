import React from 'react';
import {connect} from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
		{
			props.expenses.length === 0 ? (
				<p>No Expenses</p>
			) : (
				props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)
			)
		}
    </div>
);

const mapStateToProps = (state) => {
	return {
        expenses: selectExpenses(state.expenses, state.filters)
	};
};

export default connect(mapStateToProps)(ExpenseList);	//connect returns hoc

//when you connect a component(here ExpenseList) to a redux store, it gets rerendered as the 
//store changes with new values. So, we don't really need to subscribe to store..necessary rerendering 
//for changes is handled by react-redux. This makes our component really simple.
