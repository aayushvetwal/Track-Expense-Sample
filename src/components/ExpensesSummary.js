import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
	<div className='page-header'>
		<div className='content-container'>
			<h1 className='page-header__title'>
				Viewing <span>{props.expensesCount}</span> expense(s) totalling <span>{numeral(props.expensesTotal).format('$0,0.00')}</span>
				<div className='page-header__actions'>
					<Link className='button' to='/create'>Add Expense</Link>
				</div>
			</h1>
		</div>
	</div>
);

const matchStateToProps = (state) => {
	const expenses = selectExpenses(state.expenses, state.filters);
	return {
		expensesCount: expenses.length,
		expensesTotal: selectExpensesTotal(expenses)
	}
};

export default connect(matchStateToProps)(ExpensesSummary);