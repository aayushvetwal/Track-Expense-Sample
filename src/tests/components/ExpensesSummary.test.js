import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with one expense', () => {
	const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={100} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expensesCount={20} expensesTotal={2000} />);
	expect(wrapper).toMatchSnapshot();
});