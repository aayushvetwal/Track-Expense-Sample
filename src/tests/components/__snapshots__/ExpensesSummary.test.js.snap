// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    20
     expense(s) totallingss 
    $2,000.00
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense(s) totallingss 
    $100.00
  </p>
</div>
`;
